src/machine.rs

```diff
@@ -1371,22 +1387,29 @@ impl<'tcx> Machine<'tcx> for MiriMachine<'tcx> {
         _tcx: TyCtxtAt<'tcx>,
         machine: &Self,
         alloc_extra: &AllocExtra<'tcx>,
-        _ptr: Pointer,
+        ptr: Pointer,
         (alloc_id, prov_extra): (AllocId, Self::ProvenanceExtra),
         range: AllocRange,
     ) -> InterpResult<'tcx> {
         if machine.track_alloc_accesses && machine.tracked_alloc_ids.contains(&alloc_id) {
             machine
                 .emit_diagnostic(NonHaltingDiagnostic::AccessedAlloc(alloc_id, AccessKind::Read));
         }
-        if let Some(data_race) = &alloc_extra.data_race {
-            data_race.read(alloc_id, range, NaReadType::Read, None, machine)?;
-        }
         if let Some(borrow_tracker) = &alloc_extra.borrow_tracker {
             borrow_tracker.before_memory_read(alloc_id, prov_extra, range, machine)?;
         }
+        if let Some(genmc_ctx) = machine.concurrency_handler.as_genmc_ref() {
+            // We throw away the value we get from GenMC for non-atomic accesses, these are still handled by Miri
+            let _value = genmc_ctx.memory_load(machine, ptr.addr(), range.size)?;
+            return interp_ok(());
+        }
+
```

Comment:
Why is there an empty line here? Empty lines should indicate some sort of semantic grouping within the function.